tests/test_logic.py

```diff
@@ -1031,13 +1029,13 @@ def test_Controller_invalidate_token(mocker, homedir, session_maker):
 
 def test_Controller_logout_with_pending_replies(mocker, session_maker, homedir, reply_status_codes):
     """
-    Ensure draft reply fails on logout and that the reply_failed signal is emitted.
+    Ensure pending replies not currently being processed are marked failed and emit the
+    "reply_failed" signal.
     """
     co = Controller("http://localhost", mocker.MagicMock(), session_maker, homedir, None)
     co.api_job_queue = mocker.MagicMock()
     co.api_job_queue.stop = mocker.MagicMock()
     co.call_api = mocker.MagicMock()
-    reply_failed_emissions = QSignalSpy(co.reply_failed)
```

Comment:
The use of the `QSignalSpy` syntax was intentional (it is the canonical way of testing `QSignals` emissions). I've been converting the tests when I touch them. In this case, I don't think there is any reason not to use it, I don't see that the `mock` is testing anything that wasn't tested before, unless I'm missing something?

In case you wondered, `reply_failed_emissions` is not really a list, but it behaves like a list with regard to `len` and the bracket notation. Each item it a list of the arguments that were used during the call to `emit()`.

Reply:
I have, more controversially, simply *deleted* this test for the reasons I describe in 3333199192ae64220ef3fa52b5c82547f19383a4.  Up for debate *iff* you're willing to pair with me on reconstructing it.  ;-)

Reply:
Yah, deleting seems reasonable. Thanks for the detailed commit message! :+1: